x/supplier/keeper/prune_supplier_service_config_history.go

```diff
@@ -25,7 +25,20 @@ func (k Keeper) EndBlockerPruneSupplierServiceConfigHistory(
 
 	logger := k.Logger().With("method", "PruneSupplierServiceConfigHistory")
 
-	for _, supplier := range k.GetAllSuppliers(ctx) {
+	allSuppliersIterator := k.GetAllSuppliersIterator(ctx)
+	defer allSuppliersIterator.Close()
+
+	for ; allSuppliersIterator.Valid(); allSuppliersIterator.Next() {
+		supplier, err := allSuppliersIterator.Value()
+		if err != nil {
+			logger.Error(fmt.Sprintf("could not get supplier from iterator: %v", err))
+			return 0, err
+		}
+		if supplier == nil {
+			logger.Error(fmt.Sprintf("unexpected nil supplier in iterator at %s", allSuppliersIterator.Key()))
+			continue
+		}
+
```

Comment:
```suggestion
		supplierValue, err := allSuppliersIterator.Value()
		if err != nil {
			logger.Error(fmt.Sprintf("could not get supplier from iterator: %v", err))
			return 0, err
		}
		if supplierValue == nil {
			logger.Error(fmt.Sprintf("unexpected nil supplier in iterator at %s", allSuppliersIterator.Key()))
			continue
		}
		// Make a copy of the supplier to avoid pointer related concurrency issues
		supplier := *supplierValue

```

Reply:
See my comment below for reference